web-api/src/persistence/postgres/workitems/getDocumentQCInboxForUser.ts

```diff
@@ -1,57 +1,77 @@
+import {
+  RawWorkItemWithCaseAndDocketEntryInfo,
+  WorkItemWithCaseInfoKysely,
+} from '@web-api/persistence/postgres/workitems/schema';
+import { getDocketEntriesByDocketNumberAndDocketEntryId } from '@web-api/persistence/postgres/docketEntries/getDocketEntriesByDocketNumberAndDocketEntryId';
+import { NotFoundError } from '@web-api/errors/errors';
 
 export const getDocumentQCInboxForUser = async ({
   userId,
 }: {
   userId: string;
-}): Promise<WorkItemWithCaseInfo[]> => {
-  const workItems = await getDbReader(reader => {
+}): Promise<RawWorkItemWithCaseAndDocketEntryInfo[]> => {
+  const workItems: WorkItemWithCaseInfoKysely[] = await getDbReader(reader => {
     return workItemQCQueryBase(reader)
       .where('w.assigneeId', '=', userId)
       .where('w.completedAt', 'is', null)
       .limit(5000)
       .execute();
   });
-
-  return workItems.map(toWorkItemWithCaseInfo);
-};
-
-export type WorkItemWithCaseInfo = Omit<RawWorkItem, 'docketEntry'> & {
-  caseTitle?: string;
-  caseStatus?: string;
-  leadDocketNumber?: string;
-  trialDate?: string;
-  trialLocation?: string;
-  docketEntry: RawDocketEntry;
+  return await attachDocketEntriesToWorkItemQC({ workItems });
 };
 
 export const workItemQCQueryBase = (dbReader: Kysely<Database>) => {
   return dbReader
     .selectFrom('dwWorkItem as w')
     .leftJoin('dwCase as c', 'c.docketNumber', 'w.docketNumber')
-    .innerJoin('dwDocketEntry as d', join =>
-      join
-        .onRef('d.docketEntryId', '=', 'w.docketEntryId')
-        .onRef('d.docketNumber', '=', 'w.docketNumber'),
-    )
     .selectAll('w')
-    .select(eb => [
-      jsonObjectFrom(
-        eb
-          .selectFrom('dwDocketEntry as de')
-          .selectAll()
-          .whereRef('de.docketEntryId', '=', 'w.docketEntryId')
-          .whereRef('de.docketNumber', '=', 'w.docketNumber'),
-      ).as('docketEntry'),
+    .select([
       'c.status',
       'c.caption',
       'c.leadDocketNumber',
       'c.trialDate',
       'c.trialLocation',
     ]);
 };
+
+export const attachDocketEntriesToWorkItemQC = async ({
+  workItems,
+}: {
+  workItems: WorkItemWithCaseInfoKysely[];
+}) => {
+  if (workItems.length === 0) {
+    return [];
+  }
+
+  const docketEntryInfo = workItems.map(w => ({
+    docketEntryId: w.docketEntryId,
+    docketNumber: w.docketNumber,
+  }));
+  const docketEntries = await getDocketEntriesByDocketNumberAndDocketEntryId({
+    docketNumbersAndIds: docketEntryInfo,
+  });
+
+  const entryById = new Map<string, RawDocketEntry>();
+  for (const docketEntry of docketEntries) {
+    entryById.set(docketEntry.docketEntryId, docketEntry);
+  }
+
+  const workItemsWithDocketEntries = workItems.map(w => {
+    const docketEntry = entryById.get(w.docketEntryId);
+    if (!docketEntry) {
+      throw new NotFoundError(
+        `Could not find docket entry for work item ${w.docketNumber}, docket entry id ${w.docketEntryId}`,
+      );
+    }
```

Comment:
I'm not actually sure we want this. Might be safer to have {} as RawDocketEntry.